core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -364,18 +362,37 @@ boolean forceRefresh() {
             executor,
             metadataFuture,
             sslContextFuture);
-    refreshFuture.addListener(
-        () -> {
-          synchronized (instanceDataGuard) {
-            // update currentInstanceData with the most recent results
-            currentInstanceData = refreshFuture;
-            // schedule a replacement before the SSLContext expires;
-            nextInstanceData = executor
-                .schedule(this::performRefresh, secondsUntilRefresh(),
-                    TimeUnit.SECONDS);
+    Futures.addCallback(refreshFuture,
+        new FutureCallback<InstanceData>() {
+          public void onSuccess(InstanceData instanceData) {
+            synchronized (instanceDataGuard) {
+              // update currentInstanceData with the most recent results
+              currentInstanceData = refreshFuture;
+              // schedule a replacement before the SSLContext expires;
+              nextInstanceData = executor
+                  .schedule(() -> performRefresh(),
+                      secondsUntilRefresh(),
+                      TimeUnit.SECONDS);
+            }
           }
-        },
-        executor);
+
+          public void onFailure(Throwable t) {
+            logger.log(Level.WARNING,
+                "An error occurred while performing refresh. Retrying immediately.", t);
```

Comment:
Can we include the error in this log, so it's at least surfaced somewhere?

Reply:
I thought that was what including the throwable as the third argument did.